File: src/main/java/com/ptaf/ai/validation/AllowedYamlGuardResult.java
package com.ptaf.ai.validation;

import java.util.List;

public record AllowedYamlGuardResult(
        boolean passed,
        List<String> allowedKeysUsed,
        List<String> unknownKeysUsed,
        List<String> missingKeysDeclared,
        List<String> missingKeysUsedInFeature,
        List<String> warnings,
        List<String> blockingErrors
) {
}

File: ptaf/ai/validation/allowed_yaml_guard_result.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class AllowedYamlGuardResult:
    passed: bool
    allowed_keys: list[str]
    unknown_keys_used: list[str]
    missing_keys_declared: list[str]
    missing_keys_used_in_feature: list[str]
    warnings: list[str]
    blocking_errors: list[str]
